### PR TITLE
Log major choices to a separate log file

### PR DESCRIPTION
**Pull request by elipup**
Solving #7

### CODE REVIEW

File: logging.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 
 LOG_FILE=$(pwd)/install-$DATE.log
+CHOICE_FILE=$(pwd)/choice-$DATE.log
```
Comment (gilmoskowitz) [inline logging.sh]: Why distinguish between `LOG_FILE` and `CHOICE_FILE`? This makes it hard to align events with the user's selections.

File: logging.sh [hunk 2]
```diff
@@ -13,6 +14,15 @@ log() {
...
 
+log_choice() {
+    echo -n "$1 " >> $CHOICE_FILE
+    "$@"
+}
+
+log_arg() {
+	echo "$@" >> $CHOICE_FILE
+}
+
```
Comment (gilmoskowitz) [inline logging.sh]: Logging should include timestamps.

File: database.sh [hunk 3]
```diff
@@ -27,18 +27,18 @@ database_menu() {
...
         elif [ $RET -eq 0 ]; then
             case "$DBM" in
-            "1") set_database_info ;;
-            "2") clear_database_info ;;
-            "3") backup_database ;;
-            "4") list_databases ;;
+            "1") log_choice set_database_info ;;
+            "2") log_choice clear_database_info ;;
+            "3") log_choice backup_database ;;
+            "4") log_choice list_databases ;;
```
Comment (gilmoskowitz) [inline database.sh]: Distinguishing between selecting between action items and submenu items has two effects:

- The maintainer needs to remember when to `log_choice` and when not to
- It's harder to trace the path followed by the user through the program

Logging _every_ function entry and exit has the same net effect.

**Comment by elipup** [comment]
The choice file is a list of the action items with their arguments.  It's meant to be a file that can be "replayed" to repeat all of the actions a user performed.
The log file has the time stamps and includes the path the user took through the program.  The menu functions log when they are opened by the user.

This is what I gathered from talking with @davidbeauchamp some months ago about this issue.  We might want something different now.

**Pull request merged**